lib/HeroBonus.h

```diff
@@ -331,6 +330,21 @@ class DLL_LINKAGE CAddInfo : public std::vector<si32>
 	BONUS_NAME(SPECIAL_FIXED_VALUE_ENCHANT) /*specialty spell like Melody has, constant spell effect (i.e. 3 luck), additionalInfo = value to fix.*/\
 	BONUS_NAME(TOWN_MAGIC_WELL) /*one-time pseudo-bonus to implement Magic Well in the town*/\
 	BONUS_NAME(LIMITED_SHOOTING_RANGE) /*limits range of shooting creatures, doesn't adjust any other mechanics (half vs full damage etc). val - range in hexes, additional info - optional new range for broken arrow mechanic */\
+	BONUS_NAME(LEARN_BATTLE_SPELL_CHANCE) /*skill-agnostic eagle eye chance. subtype = 0 - from enemy, 1 - TODO: from entire battlefield*/\
+	BONUS_NAME(LEARN_BATTLE_SPELL_LEVEL_LIMIT) /*skill-agnostic eagle eye limit, subtype - school (-1 for all), others TODO*/\
+	BONUS_NAME(PERCENTAGE_DAMAGE_BOOST) /*skill-agnostic archery and offence, subtype is 0 for offence and 1 for archery*/\
+	BONUS_NAME(LEARN_MEETING_SPELL_LIMIT) /*skill-agnostic scholar, subtype is -1 for all, TODO for others (> 0)*/\
+	BONUS_NAME(ROUGH_TERRAIN_DISCOUNT) /*skill-agnostic pathfinding*/\
+	BONUS_NAME(WANDERING_CREATURES_JOIN_BONUS) /*skill-agnostic diplomacy*/\
+	BONUS_NAME(BEFORE_BATTLE_REPOSITION) /*skill-agnostic tactics, bonus for allowing tactics*/\
+	BONUS_NAME(BEFORE_BATTLE_REPOSITION_BLOCK) /*skill-agnostic tactics, bonus for blocking opposite tactics. For now donble side tactics is TODO.*/\
+	BONUS_NAME(HERO_EXPERIENCE_GAIN_PERCENT) /*skill-agnostic learning, and we can use it as a global effect also*/\
+	BONUS_NAME(UNDEAD_RAISE_CHANCE) /*Chance to raise creatures after battle as undead*/\
+	BONUS_NAME(KNOWLEDGE_TO_MANA_RATIO) /*Percentage rate of translating 10 hero knowledge to mana, used to intelligence and global bonus*/\

```

Comment:
MANA_PER_KNOWLEDGE maybe?
Also, comment is outdated - "10 hero knowledge to mana"

Reply:
@krs0 suggested KNOWLEDGE_TO_MANA_RATIO. What is better? Comment will be changed.

Reply:
Otherwise it`s impossible to accurately emulate Intelligence skill from OH3. At Basic level, it gains +25% Max MP.

Reply:
I like MANA_PER_KNOWLEDGE more since it clearly indicates dependency:
amount of mana depends on knowledge.

KNOWLEDGE_TO_MANA_RATIO does not offers that. Does this bonus affects knowledge or mana?

@Alucard648 that part is taken care of - Intelligence give 25% to base value.

Reply:
@Alucard648 it gains not 25% to max MP, but 25% to base MP.  And it works.

Reply:
Renamed.